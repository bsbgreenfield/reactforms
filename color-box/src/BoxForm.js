import React, { useState } from "react";
import {v4 as uuid} from "uuid"

function BoxForm({updateBoxes, allBoxes}) {
    const INITIAL_STATE = {
        width: "",
        height: "",
        color: "#000000",
        id: ""
    }
    const [formData, setFormData] = useState(INITIAL_STATE)
    const handleChange = (e) => {
        const { name, value } = e.target
        setFormData(formData => (
            { ...formData, [name]: value }
        ))
    }
    const handleSubmit = (e) => {
        e.preventDefault()
        if (formData.width != "" && formData.height != ""){
            formData.id = uuid()
            updateBoxes(formData)
            setFormData(INITIAL_STATE)
        }
    }
    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="width">Width</label>
            <input
                type="range"
                name="width"
                id="width"
                onChange={handleChange}
                value={formData["width"]}
                >
              
            </input>
            <label htmlFor="height">Height</label>
            <input
                type="range"
                name="height"
                id="height"
                onChange={handleChange}
                value={formData["height"]}>

            </input>
            <label htmlFor="color">Color</label>
            <input
                type="color"
                name="color"
                id="color"
                onChange={handleChange}
                value={formData["color"]}
            ></input>
            <button type="submit">Create Box!</button>
        </form>
    )
}

export default BoxForm