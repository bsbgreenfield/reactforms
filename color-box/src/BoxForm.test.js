import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxForm from "./BoxForm";

it("loads without crashing", function(){
    render(<BoxForm/>)
})

it("matches snapshot", function(){
    const {asFragment} = render(<BoxForm/>)
    expect(asFragment()).toMatchSnapshot()
})