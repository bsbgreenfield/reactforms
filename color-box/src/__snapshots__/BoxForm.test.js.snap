// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="width"
    >
      Width
    </label>
    <input
      id="width"
      name="width"
      type="range"
      value=""
    />
    <label
      for="height"
    >
      Height
    </label>
    <input
      id="height"
      name="height"
      type="range"
      value=""
    />
    <label
      for="color"
    >
      Color
    </label>
    <input
      id="color"
      name="color"
      type="color"
      value="#000000"
    />
    <button
      type="submit"
    >
      Create Box!
    </button>
  </form>
</DocumentFragment>
`;
